src/core/server/saved_objects/mappings/kibana_index_mappings_mixin.ts

```diff
@@ -17,51 +17,53 @@
  * under the License.
  */
 
+import { Server } from 'hapi';
 import { IndexMappings } from './index_mappings';
+import { EsMappings } from './lib/';
 
 /**
- *  The default mappings used for the kibana index. This is
- *  extended via uiExports type "mappings". See the kibana
- *  and timelion plugins for examples.
- *  @type {EsMappingDsl}
+ *  The default mappings used for the Kibana index. This is
+ *  extended via uiExports type "mappings". See the Kibana
+ *  and Timelion plugins for examples.
+ *  @type {EsMappings}
  */
-const BASE_SAVED_OBJECT_MAPPINGS = {
+const BASE_SAVED_OBJECT_MAPPINGS: EsMappings = {
   doc: {
-    dynamic: 'strict',
+    dynamic: 'strict' as 'strict',
     properties: {
       type: {
-        type: 'keyword'
+        type: 'keyword',
       },
       updated_at: {
-        type: 'date'
+        type: 'date',
       },
       config: {
         dynamic: true,
         properties: {
           buildNum: {
-            type: 'keyword'
-          }
-        }
+            type: 'keyword',
+          },
+        },
       },
-    }
-  }
+    },
+  },
 };
 
-export function kibanaIndexMappingsMixin(kbnServer, server) {
+export function kibanaIndexMappingsMixin(kbnServer: any, server: Server) {
```

Comment:
Not sure if `any` is appropriate here. Thoughts?

Reply:
I think I'd prefer that the mixin used for the legacy Kibana server stay with the legacy Kibana server source, and probably stay in JavaScript. It cuold be responsible for pulling in the `IndexMappings` class from the new platform, but putting legacy server mixins in the new platform feels like the wrong separation. @azasypkin how do you expect things like this to be handled in the new platform server?

Reply:
Agree!